api/segments/models.py

```diff
@@ -169,6 +170,10 @@ def does_identity_match(
                     return self.check_float_value(trait.float_value)
                 elif trait.value_type == BOOLEAN:
                     return self.check_boolean_value(trait.boolean_value)
+                elif semver.VersionInfo.isvalid(
```

Comment:
I wonder if we should have a different way of detecting if a value is semver. Perhaps we can speak to Kyle and how it might work on the FE - could he encode something into the payload for example? I'm just concerned that probably a large majority of our traits are strings so we're adding an extra parsing step for each trait value / segment condition evaluation.

Reply:
Is it just counting the number of `.`? I would say this is a few microseconds tops?

Reply:
I don't see how we would do it otherwise? Would have to hack something in unless we used a new datatype which seems like a lot of work for not much benefit.